### PR TITLE
Support customized link targets via `data-toc-href` heading attributes

### PR DESCRIPTION
**Pull request by nchilada**
In combination with `style="display: none"`, this allows users to create heading aliases that only appear in the table of contents and simply redirect to a different heading.

Something like this is needed for the [Optimizely dev docs](https://github.com/optimizely/developers.optimizely.com).  We're replacing some old headings with new headings, and we want the old heading text to appear in the table of contents as long as they can link directly to the new headings.

### CODE REVIEW

File: src/js/parse-content.js [hunk 1]
```diff
@@ -29,11 +29,12 @@ module.exports = function parseContent(options) {
   /**
    * Get important properties from a heading element and store in a plain object.
    * @param {HTMLElement} heading
+   * @param {String} tocHrefAttribute
    * @return {Object}
    */
-  function getHeadingObject(heading) {
+  function getHeadingObject(heading, tocHrefAttribute) {
     var obj = {
-      id: heading.id,
```
Comment (tscanlin) [inline src/js/parse-content.js]: Not setting the id will break the default hash handling of the browser.

Your way will work for users who click in the Toc to jump to a section. And it should work so the proper section is highlighted as you scroll. But what about back / forward buttons within a page? I think this breaks that functionality when the tocHrefAttribute is used?

**Comment by tscanlin** [comment]
Thanks for the PR, I think I understand what you're trying to do now.

Since you are going to have to change markup anyway, I think there is a way to accomplish what you want without modifying tocbot or breaking the native hash navigation behavior that browsers already have.

Can you try this in the "Try it now section", here: http://tscanlin.github.io/tocbot/

```
# Title

Tocbot builds a table of contents (TOC) from headings in an HTML document. This is useful for documentation websites or long markdown pages because it makes them easier to navigate. This library was inspired by Tocify, the main difference is that Tocbot uses native DOM methods and avoids the jQuery & jQuery UI dependencies.

## Next heading
<h2 id="alternate-next-heading" class="js-toc-ignore" style="visibility: hidden; position: relative; top: -3.2em;">Alternate next heading</h2>
```

Would the above accomplish the desired result for the Dev Docs?

**Comment by tscanlin** [comment]
This change doesn't seem broadly applicable to other people using this library and it does add complexity. I think there is a way to accomplish the objective of this PR without making these code changes. Feel free to make any counter arguments or we can discuss in private if you want. Thanks for the suggestion though.

**Comment by nchilada** [comment]
No worries Tim, and sorry for the late reply.

Other things came up and our entire project is on the backburner for a couple months, but I'll ping you offline when I can!

**Pull request closed without merging**